resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/ExecutorPodFactory.scala

```diff
@@ -265,9 +268,53 @@ private[spark] class ExecutorPodFactoryImpl(
     val executorPodWithNodeAffinity =
         nodeAffinityExecutorPodModifier.addNodeAffinityAnnotationIfUseful(
             executorPodWithInitContainer, nodeToLocalTaskCount)
-    new PodBuilder(executorPodWithNodeAffinity)
+
+    val (executorPodWithTempLocalDirs, executorContainerWithTempLocalDirs) =
+        if (shuffleManager.isEmpty) {
+          // If we're not using the external shuffle manager, we should use emptyDir volumes for
+          // shuffle directories since it's important for disk I/O for these directories to be
+          // performant. If the user has not provided a local directory, instead of using the
+          // Java temporary directory, we create one instead. This is because we want to avoid
+          // as much as possible mounting an emptyDir which overlaps with an existing path in
+          // the Docker image, which is very likely what would happen if we tried to mount the
+          // volume at Java's temporary directory path, which is /tmp in many JDKs.
+          val resolvedLocalDirs = sparkConf.get(
+              "spark.local.dir", s"$GENERATED_LOCAL_DIR_MOUNT_ROOT/${UUID.randomUUID()}")
```

Comment:
If a user has set `spark.local.dir` they may have a reason for placing the local directory in a particular mount path.  Maybe we should only mount the dirs from `spark.local.dir` as EmptyDir volumes if it's the default value, rather than user-specified?

Reply:
One use case for wanting to mount if they are user specified is if they specified multiple local directories where the backend can delegate the volumes to mount to different disks. This is something we talked about in #260.